### PR TITLE
Require permissions for showing /ps, /audit-log and /fs

### PR DESCRIPTION
**Pull request by fgoerner**
This is necessary for a customer, as they want to have users without access to /ps, /audit-log and /fs
For /ps and /audit-log hiding the links is enough, no sensitive information is show there (because the users don't have any other permissions)
For /fs, read and write access will be restricted in the customer config

* Introduces 3 new permissions (`"permission-view-files", "permission-view-processes", "permission-view-audit-log"`). These permissions are granted to all logged-in users by default.

### CODE REVIEW

File: src/main/resources/default/extensions/biz-tycho-menu/menu-jobs.html.pasta [hunk 1]
```diff
@@ -7,7 +7,7 @@
             <t:dropdownItem labelKey="JobsController.jobs" url="/jobs"
                             permission="permission-execute-jobs"
                             framework="biz.jobs"/>
-            <t:dropdownItem labelKey="Process.plural" url="/ps"
+            <t:dropdownItem labelKey="Process.plural" url="/ps" permission="permission-system-protocols"
```
Comment (andyHa) [inline src/main/resources/default/extensions/biz-tycho-menu/menu-jobs.html.pasta:10]: Where is this permission granted?

Comment (fgoerner) [reply src/main/resources/default/extensions/biz-tycho-menu/menu-jobs.html.pasta:10]: Currently it's only granted with the `flag-system-administrator` permission. Should I make it grantable by itself?

**Comment by andyHa** [comment]
Please:
* introduce 3 distinct permissions (permission-view-files, permission-view-processes, permissions-view-audit-log)
* explain their purose properly in component-biz.conf (security.permissions)
* enforce them in the appropriate controllers
* grant them to all users of the defaul scope, so that the default settings remain unchanged. Then, where appropriate, we can change these in a per-product / per-customer level. (Binding all this to system tenant only makes no sense at all as a default, as nearly every system becomes unusable without files and processes. And hiding the audit-log is a security relevant change and should be discussed as well)

**Review by meg1502 — CHANGES_REQUESTED** [review]
see review of @andyHa

**Comment by fgoerner** [comment]
Thanks for the suggestions, I introduced the permissions and set them in the respective templates and controllers.
For the audit-log we could ask the customer if it's really necessary for them to hide it.

File: src/main/java/sirius/biz/protocol/AuditLogController.java [hunk 2]
```diff
@@ -31,13 +32,19 @@
      */
     public static final String PERMISSION_AUDIT_LOGS = "permission-audit-logs";
 
+    /**
+     * Names the permissions required to view the audit log page.
+     */
+    private static final String PERMISSION_VIEW_AUDIT_LOGS = "permissions-view-audit-log";
```
Comment (sabieber) [inline src/main/java/sirius/biz/protocol/AuditLogController.java:38]:
```suggestion
    private static final String PERMISSION_VIEW_AUDIT_LOGS = "permission-view-audit-log";
```

File: src/main/resources/component-biz.conf [hunk 3]
```diff
@@ -1294,6 +1295,9 @@ security {
         permission-execute-jobs         : "Required to execute jobs"
         permission-manage-scheduler     : "Required to plan and edit scheduled jobs"
         permission-system-scripting     : "Required to execute administrator level scripts on all nodes of the cluster"
+        permission-view-files           : "Required to view files stored in the system"
+        permission-view-processes       : "Required to view processes"
+        permissions-view-audit-log      : "Required to view the audit log"
```
Comment (sabieber) [inline src/main/resources/component-biz.conf:1300]:
```suggestion
        permission-view-audit-log      : "Required to view the audit log"
```

File: src/main/resources/default/extensions/biz-tycho-menu/menu-protcols.html.pasta [hunk 4]
```diff
@@ -4,7 +4,7 @@
 <i:switch test="@point">
     <i:block name="user">
         <t:dropdownSection>
-            <t:dropdownItem labelKey="AuditLogEntry.plural" url="/audit-log" framework="biz.protocols"/>
+            <t:dropdownItem labelKey="AuditLogEntry.plural" url="/audit-log" permission="permissions-view-audit-log" framework="biz.protocols"/>
```
Comment (sabieber) [inline src/main/resources/default/extensions/biz-tycho-menu/menu-protcols.html.pasta:7]:
```suggestion
            <t:dropdownItem labelKey="AuditLogEntry.plural" url="/audit-log" permission="permission-view-audit-log" framework="biz.protocols"/>
```
```suggestion
            <t:dropdownItem labelKey="AuditLogEntry.plural" url="/audit-log" permission="permissions-view-audit-log" framework="biz.protocols"/>
```

**Review by sabieber — CHANGES_REQUESTED** [review]

**Review by ymo-sci — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/resources/component-biz.conf [hunk 5]
```diff
@@ -1283,6 +1283,7 @@ security {
         fallback-language = "en"
         known-languages = ["de", "en", "fr", "nl", "it", "es", "pl", "cs", "hu", "da", "sv", "ro", "ru", "sk", "bg"]
         display-languages = ["de", "en", "fr", "nl", "it", "es", "pl", "cs", "hu", "da", "sv", "ro", "ru", "sk", "bg"]
+        defaultRoles = ["permission-view-files", "permission-view-processes", "permission-view-audit-log"]
```
Comment (andyHa) [inline src/main/resources/component-biz.conf:1286]: please doublecheck that these are not granted to _not logged in_ users

File: src/main/resources/component-biz.conf [hunk 6]
```diff
@@ -1300,6 +1301,9 @@ security {
         permission-execute-jobs         : "Required to execute jobs"
         permission-manage-scheduler     : "Required to plan and edit scheduled jobs"
         permission-system-scripting     : "Required to execute administrator level scripts on all nodes of the cluster"
+        permission-view-files           : "Required to view files stored in the system"
+        permission-view-processes       : "Required to view processes"
+        permission-view-audit-log      : "Required to view the audit log"
```
Comment (andyHa) [inline src/main/resources/component-biz.conf:1306]:
```suggestion
        permission-view-audit-log       : "Required to view the audit log"
```

File: src/main/resources/component-biz.conf (hunk 5 above)
Comment (andyHa) [reply src/main/resources/component-biz.conf:1286]: Shouldn't we grant them using a profile via logged-in {...} using the same approach in a product, these permissions can be revoked again

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/resources/component-biz.conf (hunk 5 above)
Comment (fgoerner) [reply src/main/resources/component-biz.conf]: You mean like this? I'm not sure im using the profile correctly

File: src/main/resources/component-biz.conf [hunk 7]
```diff
@@ -1418,7 +1425,7 @@ security {
         #The 'tenant' scope used by the tenants framework (TenantUserManager etc.)
         tenant {
             # the packages of the scope
-            packages = []
+            packages = ["logged-in"]
```
Comment (sabieber) [inline src/main/resources/component-biz.conf:1428]:
```suggestion
            packages = []
```
Think this change is not required

File: src/main/resources/component-biz.conf [hunk 8]
```diff
@@ -1325,6 +1329,13 @@ security {
 
     profiles {
 
+        logged-in {
```
Comment (sabieber) [inline src/main/resources/component-biz.conf:1332]:
```suggestion
        flag-logged-in {
```
`logged-in` is only a mapped permission name in S2, the sirius generic one is `flag-logged-in`

**Review by sabieber — CHANGES_REQUESTED** [review]

**Review by sabieber — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

File: src/main/java/sirius/biz/protocol/AuditLogController.java [hunk 9]
```diff
@@ -31,13 +32,19 @@
      */
     public static final String PERMISSION_AUDIT_LOGS = "permission-audit-logs";
 
+    /**
+     * Names the permissions required to view the audit log page.
+     */
+    private static final String PERMISSION_VIEW_AUDIT_LOGS = "permission-view-audit-log";
+
```
Comment (meg1502) [inline src/main/java/sirius/biz/protocol/AuditLogController.java:39]: whats the difference to the permission above? can you expand the javadoc so this is clear?

**Review by meg1502 — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/protocol/AuditLogController.java (hunk 9 above)
Comment (fgoerner) [reply src/main/java/sirius/biz/protocol/AuditLogController.java:39]: I tried to clear it up so the permissions are less confusing. `permission-audit-logs` allows the user to see not only his own audit log, but all audit log entries of the user's tenant. `permission-view-audit-logs` allows the user to view the audit-log page at all.

**Review by meg1502 — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**